addons/website_slides/models/slide_channel.py

```diff
@@ -459,6 +461,28 @@ def _get_default_enroll_msg(self):
     def _compute_enroll(self):
         self.filtered(lambda channel: channel.visibility == 'members').enroll = 'invite'
 
+    @api.depends('visibility', 'is_member')
+    @api.depends_context('uid')
+    def _compute_is_visible_on_website(self):
+        for channel in self:
+            if channel.visibility == 'public' or channel.is_member or (not self.env.user._is_public() and channel.visibility == 'connected'):
+                channel.is_visible_on_website = True
+            else:
+                channel.is_visible_on_website = False
+
+    @api.model
+    def _search_is_visible_on_website(self, operator, value):
+        if operator not in ['=', '!='] or not isinstance(value, bool):
+            raise NotImplementedError(_('Operation not supported'))
+        check_is_visible_on_website = operator == '=' and value or operator == '!=' and not value
+        domain = [('is_member', '=', True)]
+        if not self.env.user._is_public():
+            domain = expression.OR([domain, [('visibility', 'in', ['public', 'connected'])]])
+        else:
+            domain = expression.OR([domain, [('visibility', '=', 'public')]])
```

Comment:
Actually if the user is not connected, it cannot be member so the first clause can be removed.